peer.go

```diff
@@ -88,48 +96,117 @@ func (b *baseSession) Close() error {
 	return b.client.NetConn().Close()
 }
 
-func NewWebSocketPeer(conn net.Conn, protocol string, binary, server bool) (Peer, error) {
-	var wsReader ReaderFunc
-	var wsWriter WriterFunc
-	var err error
-	if server {
-		wsReader, wsWriter, err = ServerSideWSReaderWriter(binary)
-	} else {
-		wsReader, wsWriter, err = ClientSideWSReaderWriter(binary)
+func NewWebSocketPeer(conn net.Conn, protocol string, binary, server bool, keepaliveInterval time.Duration,
```

Comment:
create a new WSPeerConfig struct and move all parameters of this function there (except for conn)